src/Compilers/Core/Portable/Syntax/SyntaxNode.cs

```diff
@@ -325,9 +325,10 @@ public virtual void WriteTo(TextWriter writer)
         /// <exception cref="ArgumentException"><paramref name="checksumAlgorithm"/> is not supported.</exception>
         public SourceText GetText(Encoding? encoding = null, SourceHashAlgorithm checksumAlgorithm = SourceHashAlgorithm.Sha1)
         {
-            var builder = new StringBuilder(this.Green.FullWidth);
-            this.WriteTo(new StringWriter(builder));
-            return new StringBuilderText(builder, encoding, checksumAlgorithm);
+            var fullWidth = this.Green.FullWidth;
+            var writer = SourceTextWriter.Create(encoding, checksumAlgorithm, fullWidth);
```

Comment:
this helper gets an appropriate writer depending on the length of the final source text.  For <85k (the loh limit) you get a simple string writer.  For >85k you get a chunking writer.